code/src/java/pcgen/cdom/facet/CDOMWrapperInfoFacet.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Thomas Parker, 2014.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+package pcgen.cdom.facet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import pcgen.cdom.base.CDOMObject;
+import pcgen.cdom.base.DataSetInitializedFacet;
+import pcgen.cdom.enumeration.DataSetID;
+import pcgen.cdom.facet.base.AbstractSubAssociationFacet;
+import pcgen.core.Ability;
+import pcgen.core.ArmorProf;
+import pcgen.core.Campaign;
+import pcgen.core.Deity;
+import pcgen.core.Domain;
+import pcgen.core.Equipment;
+import pcgen.core.Kit;
+import pcgen.core.Language;
+import pcgen.core.PCAlignment;
+import pcgen.core.PCCheck;
+import pcgen.core.PCClass;
+import pcgen.core.PCStat;
+import pcgen.core.PCTemplate;
+import pcgen.core.Race;
+import pcgen.core.ShieldProf;
+import pcgen.core.SizeAdjustment;
+import pcgen.core.WeaponProf;
+import pcgen.output.actor.KeyActor;
+import pcgen.output.actor.OutputNameActor;
+import pcgen.output.actor.SourceActor;
+import pcgen.output.base.OutputActor;
+import pcgen.rules.context.LoadContext;
+
+/**
+ * This Facet stores the Actors usable in Freemarker for a given DataSetID.
+ */
+public class CDOMWrapperInfoFacet
+		extends
+		AbstractSubAssociationFacet<DataSetID, Class<?>, String, OutputActor<?>>
+		implements DataSetInitializedFacet
+{
+
+	private DataSetInitializationFacet datasetInitializationFacet;
+
+	@Override
+	public void initialize(LoadContext context)
+	{
+		initialize(context.getDataSetID());
+	}
+
+	/**
+	 * Initializes this facet for the given DataSetID.
+	 * 
+	 * @param dsID
+	 *            The DataSetID for which this facet should be initialized
+	 */
+	public void initialize(DataSetID dsID)
+	{
+		set(dsID, CDOMObject.class, "source", new SourceActor());
```

Comment:
They get cleaned up.  Here is the "process":

All the facets are weak references of the Identifier (CharID or DataSetID) to their stored information, so the question can be rephrased: Does the DataSetID get appropriately cleaned up when we switch sources?

The DataSetID is "owned by" the LoadContextInst, and I have no reason to believe we have a LoadContext-binding memory leak (we had one at one point, but that has been addressed).

So the assertion becomes: Assuming no one "holds onto" the DataSetID and the LoadContext gets forgotten, the DataSetID should be cleaned up. 

Thus in my mind the remaining question is: "Are there any 'bad actors' (non-facets) that maintain a reference to the DataSetID?"  Since the only other place the DataSetID is "captured" is in the CharID (so the PC can reference back to the loaded data), any retention will also be associated with a PlayerCharacter based memory leak, which I also think we don't have at the moment.  So I believe we are appropriately managing memory in this case.